extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java

```diff
@@ -935,6 +939,46 @@ private void handleReturn(ClassInfo restClientInterface, String defaultMediaType
                 genericReturnType = createGenericTypeFromParameterizedType(methodCreator, paramType);
             }
         }
+        Integer continuationIndex = null;
+        //TODO: there should be an SPI for this
+        if (returnCategory == ReturnCategory.BLOCKING) {
+            List<Type> parameters = jandexMethod.parameters();
+            if (!parameters.isEmpty()) {
+                Type lastParamType = parameters.get(parameters.size() - 1);
+                if (lastParamType.name().equals(CONTINUATION)) {
+                    continuationIndex = parameters.size() - 1;
+                    returnCategory = ReturnCategory.COROUTINE;
+
+                    try {
+                        Thread.currentThread().getContextClassLoader().loadClass(UNI_KT.toString());
+                    } catch (ClassNotFoundException e) {
+                        //TODO: make this automatic somehow
+                        throw new RuntimeException("Suspendable rest client method" + jandexMethod + " is present on class "
+                                + jandexMethod.declaringClass()
+                                + " however io.smallrye.reactive:mutiny-kotlin is not detected. Please add a dependency on this artifact.");
+                    }
+
+                    //we infer the return type from the param type of the continuation
```

Comment:
Why do we need the type parameter here?  Can't we just look at the return type of the `suspend` function?

Reply:
ahhh.  i'm sure there's some magic there I haven't dug in to just yet, then.